keymap/vim.js

```diff
@@ -3483,7 +3483,7 @@
         },
         isComplete: function(state) {
           if (state.nextCh === '#') {
-            var token = state.lineText.match(/#(\w+)/)[1];
+            var token = state.lineText.match(/^#(\w+)/)[1];
```

Comment:
Note: This is a native `String.match` call. However, we know the string must start with the `#` because of the `nextCh === '#'` check above. As all other changes, this one does not have any effect, but improves performance.